specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2024-05-01-preview/appplatform.json

```diff
@@ -15184,6 +15691,285 @@
+            "Running",
+            "Pending",
+            "Canceled",
+            "Failed",
+            "Completed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "JobExecutionRunningState",
+            "modelAsString": true
+          }
+        },
+        "startTime": {
+          "description": "Job execution start time.",
+          "format": "date-time",
+          "type": "string"
+        },
+        "endTime": {
+          "description": "Job execution end time.",
+          "format": "date-time",
+          "type": "string"
+        },
+        "properties": {
+          "description": "Job's execution properties.",
+          "$ref": "#/definitions/JobExecutionProperties"
+        },
+        "jobSnapshot": {
+          "$ref": "#/definitions/JobResourceProperties",
+          "description": "The properties of job configuration"
+        }
+      }
+    },
+    "JobExecutionCollection": {
+      "description": "Azure Spring App Job executions collection.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of Job executions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobExecution"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "EnvVar": {
+      "description": "Azure Spring Apps components' environment variable.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Environment variable name.",
+          "type": "string"
+        },
+        "value": {
+          "description": "Non-secret environment variable value.",
+          "type": "string"
+        },
+        "secretValue": {
+          "description": "secret environment variable value.",
+          "type": "string",
+          "x-ms-secret": true
+        }
+      }
+    },
+    "EnvSecretsCollection": {
+      "description": "Secret environment variable collection.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Secret"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        }
+      }
+    },
+    "ConfigServerReference": {
+      "description": "A reference to the Config Server",
+      "required": [
+        "resourceId"
+      ],
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "Resource Id of the Config Server",
+          "type": "string",
+          "format": "arm-id",
+          "x-ms-arm-id-details": {
+            "allowedResources": [
+              {
+                "type": "Microsoft.AppPlatform/Spring/configServers"
```

Comment:
Will it support ACS later? If this is a potential option, `configurationReference` may be better.

Reply:
Yes, renamed to `ManagedComponentReference` to support ACS, Service Registry later.